snuba/datasets/dataset.py

```diff
@@ -185,12 +185,12 @@ def __init__(
     def time_expr(self, column_name: str, granularity: int, table_alias: str) -> str:
         real_column = qualified_column(column_name, table_alias)
         template = {
-            3600: "toStartOfHour({column})",
-            60: "toStartOfMinute({column})",
-            86400: "toDate({column})",
+            3600: "toStartOfHour({column}, 'Universal')",
+            60: "toStartOfMinute({column}, 'Universal')",
+            86400: "toDate({column}, 'Universal')",
         }.get(
```

Comment:
I am not sure about the intended scope of this PR. Adding Universal here will ensure that the timezone is always added when referencing bucketed time columns. There are several additional places in Snuba where we silently add a toDateTime function call. Examples
formatting literals: https://github.com/getsentry/snuba/blob/master/snuba/util.py#L234-L236
https://github.com/getsentry/snuba/blob/master/snuba/clickhouse/formatter.py#L67-L69

So my question is what are the boundaries of the problem you are trying to address ?

Reply:
I'm trying to address this issue:
https://sentry.io/organizations/sentry/issues/1514869389/?project=1&query=is%3Aunresolved+assigned%3Ame&statsPeriod=14d

I've only seen it as an issue in Discover. So I only changed the paths I know Discover touches.

Reply:
Then this seems it would work only in some cases. Isn't this supposed to be applied to every function that receives a timestamp column as parameter ?

Reply:
I suppose that's true. OK I'll search for more places to fix this.

Reply:
I think this will be fine, but still not enough. But when I said "Isn't this supposed to be applied to every function that receives a timestamp column as parameter" I meant we should probably need a query processor that replaces every DateTime column in the query into a call to toDateTime(asdasd, 'Universal'). Otherwise the result would be that an aggregation built based on any time column that is not transformed by the TimeSeriesDataset transformation, would still not work.